tests/lib/devicetree/api/app.overlay

```diff
@@ -540,6 +540,14 @@
 				ranges = <0x0 0x0 0x0 0x3eff0000 0x10000>,
 				         <0x0 0x10000000 0x0 0x10000000 0x2eff0000>;
 			};
+
+			test_ranges_empty: empty@2 {
+				reg = <0 2 1>;
+				#address-cells = <2>;
+				#size-cells = <1>;
+
+				ranges;
```

Comment:
note: maybe test the version that is not empty?

Reply:
Non-empty `ranges` are already tested. I added the empty one to verify that `DT_NODE_HAS_PROP()` returns 1 with it, which is the premise of this PR.